modules/javafx.graphics/src/main/java/com/sun/glass/ui/headless/HeadlessRobot.java

```diff
@@ -0,0 +1,406 @@
+                switch (key) {
+                    case KeyEvent.VK_0: c = ')'; break;
+                    case KeyEvent.VK_1: c = '!'; break;
+                    case KeyEvent.VK_2: c = '@'; break;
+                    case KeyEvent.VK_3: c = '#'; break;
+                    case KeyEvent.VK_4: c = '$'; break;
+                    case KeyEvent.VK_5: c = '%'; break;
+                    case KeyEvent.VK_6: c = '^'; break;
+                    case KeyEvent.VK_7: c = '&'; break;
+                    case KeyEvent.VK_8: c = '*'; break;
+                    case KeyEvent.VK_9: c = '('; break;
+                }
+            } else {
+                c = (char) (key - KeyEvent.VK_0 + '0');
+            }
+        } else if (key == KeyEvent.VK_SPACE) {
+            c = ' ';
+        } else if (key == KeyEvent.VK_TAB) {
+            c = '\t';
+        } else if (key == KeyEvent.VK_ENTER) {
+            c = (char)13;
+        } else if (key == KeyEvent.VK_MULTIPLY) {
+            c = '*';
+        } else if (key == KeyEvent.VK_DIVIDE) {
+            c = '/';
+        } else if (shifted) {
+            switch (key) {
+                case KeyEvent.VK_BACK_QUOTE: c = '~'; break;
+                case KeyEvent.VK_COMMA: c = '<'; break;
+                case KeyEvent.VK_PERIOD: c = '>'; break;
+                case KeyEvent.VK_SLASH: c = '?'; break;
+                case KeyEvent.VK_SEMICOLON: c = ':'; break;
+                case KeyEvent.VK_QUOTE: c = '\"'; break;
+                case KeyEvent.VK_BRACELEFT: c = '{'; break;
+                case KeyEvent.VK_BRACERIGHT: c = '}'; break;
+                case KeyEvent.VK_BACK_SLASH: c = '|'; break;
+                case KeyEvent.VK_MINUS: c = '_'; break;
+                case KeyEvent.VK_EQUALS: c = '+'; break;
+            }        } else {
+            switch (key) {
+                case KeyEvent.VK_BACK_QUOTE: c = '`'; break;
+                case KeyEvent.VK_COMMA: c = ','; break;
+                case KeyEvent.VK_PERIOD: c = '.'; break;
+                case KeyEvent.VK_SLASH: c = '/'; break;
+                case KeyEvent.VK_SEMICOLON: c = ';'; break;
+                case KeyEvent.VK_QUOTE: c = '\''; break;
+                case KeyEvent.VK_BRACELEFT: c = '['; break;
+                case KeyEvent.VK_BRACERIGHT: c = ']'; break;
+                case KeyEvent.VK_BACK_SLASH: c = '\\'; break;
+                case KeyEvent.VK_MINUS: c = '-'; break;
+                case KeyEvent.VK_EQUALS: c = '='; break;
+            }
+        }
+        return c == '\000' ? NO_CHAR : new char[] { c };
+    }
+
+
+    private int getKeyModifiers() {
+        int answer = 0;
+        if (this.specialKeys.keyControl) answer = answer | KeyEvent.MODIFIER_CONTROL;
+        if (this.specialKeys.keyShift) answer = answer | KeyEvent.MODIFIER_SHIFT;
+        if (this.specialKeys.keyCommand) answer = answer | KeyEvent.MODIFIER_COMMAND;
+        if (this.specialKeys.keyAlt) answer = answer | KeyEvent.MODIFIER_ALT;
```

Comment:
You can remove four utterances of the word "answer" by using the `|=` operator.

Reply:
done